android-sdk/src/test/java/com/checkout/android_sdk/logging/FramesLoggingEventsTest.kt

```diff
@@ -1,36 +1,58 @@
 package com.checkout.android_sdk.logging
 
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import com.checkout.android_sdk.Utils.CheckoutTheme
 import com.checkout.android_sdk.Utils.Environment
+import com.checkout.android_sdk.Utils.filterNotNullValues
 import com.checkout.eventlogger.domain.model.MonitoringLevel
 import org.junit.Assert.assertEquals
+import org.junit.Before
 import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
 import java.util.Locale
 
+@RunWith(RobolectricTestRunner::class)
 class FramesLoggingEventsTest {
+    private lateinit var mockContext: Context
 
-  @Test
+
+    @Before
+    internal fun setUp() {
+        mockContext = ApplicationProvider.getApplicationContext()
+    }
+
+    @Test
     fun `test PaymentFormPresentedEvent map to correct values`() {
 
         assertEquals(getExpectedPaymentFormPresentedLoggingEvent().typeIdentifier,
-            FramesLoggingEventDataProvider.logPaymentFormPresentedEvent().typeIdentifier)
+            FramesLoggingEventDataProvider.logPaymentFormPresentedEvent(mockContext).typeIdentifier)
 
         assertEquals(getExpectedPaymentFormPresentedLoggingEvent().monitoringLevel,
-            FramesLoggingEventDataProvider.logPaymentFormPresentedEvent().monitoringLevel)
+            FramesLoggingEventDataProvider.logPaymentFormPresentedEvent(mockContext).monitoringLevel)
 
         assertEquals(getExpectedPaymentFormPresentedLoggingEvent().properties,
-            FramesLoggingEventDataProvider.logPaymentFormPresentedEvent().properties)
+            FramesLoggingEventDataProvider.logPaymentFormPresentedEvent(mockContext).properties)
 
     }
 
 
     private fun getExpectedPaymentFormPresentedLoggingEvent(): FramesLoggingEvent {
```

Comment:
What is the use of this test helper function when if looks exact similar to your core logic of logPaymentFormPresentedEvent ?

Reply:
So, we are mapping properties of events like theme and local device language with production environment to method name as getExpectedPaymentFormPresentedLoggingEvent's property which is created using mocking context to validate the production environment's properties.

Reply:
Can we avoid writing additional code in test to make the test work ? instead use the logic within the SDK as its repeating in logPaymentFormPresentedEvent.